### PR TITLE
conda-package-handling v1.5.0

### PR DESCRIPTION
**Pull request by msarahan**
Merge only after success.

This pull request was auto-generated by [rever](https://regro.github.io/rever-docs/)

### CODE REVIEW

**Comment by msarahan** [comment]
vs2008 seems like it is destined to fail right now.  I don't think conda-forge zstd supports vs2008 (definitely not currently, not sure about past).  I'm not clear why defaults isn't getting used - strict channel priority?

**Comment by mariusvniekerk** [comment]
We should have a `1.3.8` or a `1.3.7` for `zstd` that was built with vc9.  Not sure why the solver is not finding it.

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by mariusvniekerk** [comment]
We can alter the conda-forge pinning for zstd to be zipped with vc on windows to pick and older one?

**Comment by msarahan** [comment]
@mariusvniekerk, you are right.  It is the pin.  Thanks for pointing that out.

@isuruf static linking here is being done to avoid library load errors, which have been multitudinous and bizarre.  I will try to assemble the licenses for the associated packages.

**Comment by isuruf** [comment]
> I will try to assemble the licenses for the associated packages.

Thanks. Or add a conda-build option like below if you have the time. :wink:
```yaml
about:
  bundle_license_files:
    - {{ pin_compatible("bzip2") }}
```

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# c_compiler:        # [win]
+#   - vs2015   # [win and py3k]
+#   - vs2015   # [win and py3k]
+#   - vs2008   # [win and py2k]
+# cxx_compiler:        # [win]
+#   - vs2015   # [win and py3k]
+#   - vs2015   # [win and py3k]
+#   - vs2008   # [win and py2k]
+zstd:        # [win and py2k]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml]: I don't think `py27` selector works. Why don't you hard code the version in `meta.yaml` and remove this file?

Comment (msarahan) [reply recipe/conda_build_config.yaml]: good idea!

File: recipe/static_build.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        ADDLIB libz.a
+        ADDLIB libzstd.a
+        SAVE
+        END
+EOM
+    ranlib libarchive_and_deps.a
+    popd
+fi
+
+cp -R $RECIPE_DIR/static_deps_licenses $PREFIX/libarchive_static_deps_licenses
```
Comment (isuruf) [inline recipe/static_build.sh]: Can you use the `license_file` field to list them?

**Comment by jjhelmus** [comment]
The vs2008 build on AppVeyor is not picking up the macro definitions for INT32_MAX and similar from the `stdint.h` file in the `msinttypes` package.  

Compare [VS2015](https://ci.appveyor.com/project/conda-forge/conda-package-handling-feedstock/builds/27241695/job/yf1hcse27vuoi57q#L985):
```
980-- Looking for EFTYPE - not found
981-- Looking for EILSEQ
982-- Looking for EILSEQ - found
983-- Looking for D_MD_ORDER
984-- Looking for D_MD_ORDER - not found
985-- Looking for INT32_MAX
986-- Looking for INT32_MAX - found
987-- Looking for INT32_MIN
988-- Looking for INT32_MIN - found
989-- Looking for INT64_MAX
990-- Looking for INT64_MAX - found
```

To [VS2008](https://ci.appveyor.com/project/conda-forge/conda-package-handling-feedstock/builds/27241695/job/00m0thwvefu1hukj#L938):
```
933-- Looking for EFTYPE - not found
934-- Looking for EILSEQ
935-- Looking for EILSEQ - not found
936-- Looking for D_MD_ORDER
937-- Looking for D_MD_ORDER - not found
938-- Looking for INT32_MAX
939-- Looking for INT32_MAX - not found
940-- Looking for INT32_MIN
941-- Looking for INT32_MIN - not found
942-- Looking for INT64_MAX
943-- Looking for INT64_MAX - not found
944-- Looking for INT64_MIN
945-- Looking for INT64_MIN - not found
```

I think this may be a result of the configuration not being able to find `windows.h` and other [Windows specific headers](https://ci.appveyor.com/project/conda-forge/conda-package-handling-feedstock/builds/27241695/job/00m0thwvefu1hukj#L729):
```
-- Looking for 22 include files sys/types.h, ..., windows.h
-- Looking for 22 include files sys/types.h, ..., windows.h - not found
-- Looking for 22 include files sys/types.h, ..., wincrypt.h
-- Looking for 22 include files sys/types.h, ..., wincrypt.h - not found
-- Looking for 22 include files sys/types.h, ..., winioctl.h
-- Looking for 22 include files sys/types.h, ..., winioctl.h - not found
```

**Comment by jjhelmus** [comment]
The Windows `zstd` package for vc 9 does not include a static version of of the library.  This is leading to the failed builds.

**Comment by msarahan** [comment]
@conda-forge-admin please rerender

**Comment by msarahan** [comment]
It looks to me like the static libarchive was not built for power or ARM: https://anaconda.org/conda-forge/_libarchive_static_for_cph/files

@jjhelmus was this intentional?

**Comment by jjhelmus** [comment]
> was this intentional?

aarch64 and ppc64le needs to be enabled manually in the feedstock.

**Pull request merged**